blocks/inspector-controls/text-control/index.js

```diff
@@ -8,14 +8,15 @@ import { withInstanceId } from 'components';
  */
 import './style.scss';
 
-function TextControl( { label, value, instanceId, onChange } ) {
+function TextControl( { label, value, instanceId, onChange, ...props } ) {
 	const id = 'inspector-text-control-' + instanceId;
+    const type = props.type || 'text';
```

Comment:
Could we use default values + destructuring instead?

`TextControl( { label, value, instanceId, onChange, type = 'text', ...props } ) {